src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -963,6 +974,19 @@ ExecuteModifyTasks(List *taskList, bool expectResults, ParamListInfo paramListIn
 		return 0;
 	}
 
+	foreach(taskCell, taskList)
+	{
+		Task *task = (Task *) lfirst(taskCell);
+		ShardInterval *shardInterval = LoadShardInterval(task->anchorShardId);
+		Oid relationId = shardInterval->relationId;
+
+		/*
+		 * If we are dealing with partitioned table, we also need to lock its
+		 * partitions.
+		 */
+		LockPartitionRelationsIfPartitioned(relationId, RowExclusiveLock);
```

Comment:
Is that guaranteed that all tasks belong to the same relation (i.e., anchorShardId's are always  belong to the same relation)?

I think even if that is true for today, having the loop seems better in the long term in case we'd have tasks that belong to different tables?